src/tests/multihost/ipa/test_misc.py

```diff
@@ -303,3 +303,86 @@ def test_authentication_indicators(self, multihost):
                                                  ' |tail -10')
         ssh.close()
         assert 'indicators: 2' in search.stdout_text
+
+    def test_ssh_hash_knownhosts(self, multihost, reset_password,
+                                 setup_ipa_client, backupsssdconf):
+        """
+        :title: Current value of ssh_hash_known_hosts causes error in
+         the default configuration in FIPS mode.
+        :description: In SSSD the default value for ssh_hash_known_hosts
+         is set to true, It should be changed to false for consistency with
+         the OpenSSH setting that does not hashes host names by default
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2014249
+        :id: 1cee74c8-a0ad-44d4-8287-a32e3266de22
+        :customerscenario: false
+        :steps:
+            1. Stop SSSD
+            2. Configure SSSD with ssh having default value of
+               ssh_hash_known_hosts / ssh_hash_known_hosts = True /
+               ssh_hash_known_hosts = False
+            3. Remove /var/lib/sss/pubconf/known_hosts file
+            4. Start SSSD
+            5. Perform SSH using IPA user
+            6. Check if hostnames are hashed/unhashed in
+               /var/lib/sss/pubconf/known_hosts
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+            4. Should succeed
+            5. Should succeed
+            6. Hostnames should be hashed/unhashed as per the value of
+               ssh_hash_known_hosts
+        """
+        tools = sssdTools(multihost.client[0])
+        server_host = multihost.master[0].sys_hostname
+        rm_host_keys = "rm -rf /tmp/ssh_host0003_rsa*"
+        rm_known_hosts = "rm -rf /var/lib/sss/pubconf/known_hosts"
+        view_known_hosts = "cat /var/lib/sss/pubconf/known_hosts"
+        # adding host to IPA server
+        multihost.master[0].run_command(r"ssh-keygen -q -t rsa -N '' -C '' -f /tmp/ssh_host0003_rsa")
+        multihost.master[0].run_command("ipa host-mod %s --sshpubkey="
+                                        "\"$(cat /tmp/ssh_host0003_rsa.pub)\" "
+                                        "--updatedns"
+                                        % multihost.client[0].sys_hostname)
+
+        def check_hostname_hash(hash_value=None):
+            #  no hash_value or hash_value = True or hash_value = False
+            multihost.client[0].service_sssd("stop")
+            if hash_value is None:
+                sssd_conf_cmd = multihost.client[0].\
+                    run_command("cat /etc/sssd/sssd.conf")
+                sssd_conf = str(sssd_conf_cmd.stdout_text).strip()
+                if "ssh_hash_known_hosts" in sssd_conf:
+                    ssh_section = "ssh"
+                    ssh_param = {"ssh_hash_known_hosts": ""}
+                    tools.sssd_conf(ssh_section, ssh_param, action="delete")
```

Comment:
Can't you use `action="delete"` unconditionally (without check option is present first)? What happens if you try to delete non-existent option?

Reply:
Doesn't it support "nothrow" mode/arg?
But even if it throws, it would be easier to try/catch than to check.

Reply:
The test case will be skipped upon encountering an exception therefore as discussed I'll have to stick with this.

Reply:
How does is it work in case exception is handled in place?

Reply:
Usually, it would skip from there but I found a workaround for expecting an exception and still continue the execution, I'll push it if it succeeds.

Reply:
What "it"?
If you handle exception in place it doesn't propagate anywhere.